### DISCUSSION TITLE
Respecting 429 `Retry-After` headers

### DISCUSSION THREAD

**Discussion by mivade** [prkumar/uplink#235] [Q&A] [answered]
[RFC 6585](https://datatracker.ietf.org/doc/html/rfc6585) describes the 429 status code to indicate hitting rate limits. This type of response may include a `Retry-After` header indicating the time in seconds to wait before trying again. Naively I would expect to be able to define my own backoff function to parse this header and yield that value to indicate how long to wait, but it looks like backoff functions have no knowledge of the response itself. Is there a way to get this functionality?

**Comment by prkumar** [comment]
@mivade - Thanks for posting! It looks like the builtin [`@retry`](https://github.com/prkumar/uplink/blob/7bb0b3526c9ec8f1d87274b5c7143f75d7a92fcd/uplink/retry/retry.py#L14) decorator does not support this. I've created #238 as a FR for this. As you mentioned, a possible solution is to allow the backoff function to inspect the response object.

**Comment by prkumar** [comment, marked as answer]
@mivade - I've drafted #247, which will add a way to support custom backoff functions that can inspect the response.
